hardy/src/main.rs

```diff
@@ -71,44 +74,46 @@ impl Population {
     pub fn swap_generations(&mut self) -> () {
         mem::swap(&mut self.parents, &mut self.children);
     }
-}
-
-const HOMOZYGOUS: u8 = 2;
 
-fn mate<R: Rng>(population: &mut Population, rng: &mut R) -> () {
-    let parent_range = Range::new(0usize, population.n_individuals);
-    let allele_range = Range::new_inclusive(0usize, 1usize);
-
-    for mut child in population.children {
-        child.0 = 0;
-        child.1 = 0;
-        
-        let parent1 = population.parents[parent_range.sample(rng)];
-        let parent2 = population.parents[parent_range.sample(rng)];
-
-        match parent1 {
-            Individual(HOMOZYGOUS, _) => child.0 = 1,
-            Individual(_, HOMOZYGOUS) => child.1 = 1,
-            _ => if rng.gen::<bool>() {
-                child.0 = 1;
-            } else {
-                child.1 = 1;
+    fn mate<R: Rng>(&mut self, rng: &mut R) -> () {
+        let parent_range = Range::new(0usize, self.n_individuals);
+        let allele_range = Range::new_inclusive(0usize, 1usize);
+
+        for mut child in self.children.iter_mut() {
```

Comment:
[`iter_mut()`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.iter_mut) is an iterator over mutable borrows, allowing you to modify the items.